### DISCUSSION TITLE
Standardized activity

### DISCUSSION THREAD

**Discussion by jasperjansz** [blockstack/ui#132] [Ideas]
We at Hiro build three products that show activity: the web wallet, desktop wallet, and the explorer. Currently, they all show activity in a different way:

![image](https://user-images.githubusercontent.com/25477905/132848240-12a2b431-bd63-47ee-9d85-e08184c0d886.png)
![image](https://user-images.githubusercontent.com/25477905/132848295-2a4f601e-0d05-46ad-b70d-54b2fcf4db31.png)
![image](https://user-images.githubusercontent.com/25477905/132848321-c39dd801-fdb0-4f10-95d7-24b732b3dfa9.png)

Many users use at least the web wallet and explorer, and a subset of them will use the desktop wallet too. To help avoid confusion, we should standardize how we show activity across our products where we can. 

**Proposal**
- Make the wallets share the same components — the only difference is the desktop wallet is wider.
- Reuse what we can for the explorer, but add richer information.

You can find the components in the design system [here](https://www.figma.com/file/Li7qK8ZIG9c5dKSNOPO4iCtv/%E2%9D%96-Design-System?node-id=2582%3A3171), and [here](https://www.figma.com/file/MQgs9NBbmzFcosnu9Itvp6/standardized-activity?node-id=49%3A4566) is an example of how they'd look in action.

![image](https://user-images.githubusercontent.com/25477905/132848867-a86b33fa-f082-4a05-ac71-c9ad01dbd44c.png)

Please leave your comments! It's super easy to make changes the way it's set up in Figma.

**Comment by kyranjamie** [comment]
Love it. Let's do this.

Only thing I'd note is implementation related. I don't think this belongs in the UI library. Doing so would mix domain concerns, the concept of transactions, with domain-agnostic components, buttons, boxes etc. We should create a separate monorepo for these domain-aware components, and publish them as their own npm packages.

**Comment by beguene** [comment]
Looks great ! This consistency and reusability will be really valuable.
